.zuul.yaml

```diff
@@ -3,7 +3,67 @@
     name: packit-service/packit-service
     check:
       jobs:
-        - noop
+        - packit-service-pre-commit
+        - packit-service-tests
     gate:
       jobs:
-        - noop
+        - packit-service-pre-commit
+        - packit-service-tests
+
+- import_playbook: ./files/zuul/{{ item }}-secret.yaml
+  with_items:
+  - cert.pem
+  - chain.pem
+  - copr
+  - fullchain.pem
+  - httpd-packit.conf
+  - id_rsa
+  - id_rsa.pub
+  - packit-service.yaml
+  - private-key.pem
+  - privkey.pem
+  - ssh_config
+  - fedora.keytab
+
+- job:
+    name: packit-service-pre-commit
+    parent: base
+    description: Check precommit
+    run: files/zuul-pre-commit.yaml
+    extra-vars:
+      ansible_python_interpreter: /usr/bin/python3
+    nodeset:
+      nodes:
+        - name: test-node
+          label: cloud-fedora-30
+
+- job:
+    name: packit-service-tests
+    parent: base
+    description: Run tests of packit via rpms
+    pre-run:
+      - files/install-openshift.yaml
+    run: files/zuul-tests.yaml
+    extra-vars:
+      with_testing: true
+      ansible_python_interpreter: /usr/bin/python3
+    nodeset:
+      nodes:
+        - name: test-node
+          label: cloud-fedora-30
+    secrets:
+      - name: "{{ item }}"
+        secret: "{{ item }}"
+        with_items:
+        - cert.pem
+        - chain.pem
+        - copr
+        - fullchain.pem
+        - httpd-packit.conf
+        - id_rsa
+        - id_rsa.pub
+        - packit-service.yaml
+        - private-key.pem
+        - privkey.pem
+        - ssh_config
+        - fedora.keytab
```

Comment:
Using such secrets in a check job is a security concern. A follow-up PR may be able to change the run task to leak those files. Instead, it seems like this job should be managed in a config project, like that, the job change won't be tested speculatively and it will be possible to run check test with secret (assuming the job doesn't run anything locally from the PR content). We can try something like that, in a project named `config-project` for example:

```yaml
# github.com/packit-service/config-project/zuul.yaml
---
- job:
    name:  packit-deployment-test
    required-projects:
      - packit-service/packit-deployment
    allowed-projects:
      - <name-of-project-that-can-use-the-job-in-check>
    final: true # this prevent job modification
    secrets: []
    run: playbooks/test-deployment.yaml # this file is in a config repository, preventing speculative test
```

Also, since the resources may persist accross job run, we should make sure a [cleanup](https://zuul-ci.org/docs/zuul/user/config.html#attr-job.cleanup-run) is setup.

Alternatively, zuul can provides an openshift project as a test resources if we add your cluster in nodepool.

Reply:
> Using such secrets in a check job is a security concern. A follow-up PR may be able to change the run task to leak those files. Instead, it seems like this job should be managed in a config project, like that, the job change won't be tested speculatively and it will be possible to run check test with secret (assuming the job doesn't run anything locally from the PR content). We can try something like that, in a project named `config-project` for example:
> 
> ```yaml
> # github.com/packit-service/config-project/zuul.yaml
> ---
> - job:
>     name:  packit-deployment-test
>     required-projects:
>       - packit-service/packit-deployment
>     allowed-projects:
>       - <name-of-project-that-can-use-the-job-in-check>
>     final: true # this prevent job modification
>     secrets: []
>     run: playbooks/test-deployment.yaml # this file is in a config repository, preventing speculative test
> ```
> 

I already created repository here: https://github.com/packit-service/packit-service-zuul

> Also, since the resources may persist accross job run, we should make sure a [cleanup](https://zuul-ci.org/docs/zuul/user/config.html#attr-job.cleanup-run) is setup.
> 

+1